app/controllers/messages_controller.rb

```diff
@@ -34,17 +34,17 @@ def update
   private
 
   def set_message
-    @message = Message.find(params[:id])
+    @message = @organization.messages.find(params[:id])
   end
 
   def set_request
-    request_id = params[:id] ? Message.find(params[:id]).request_id : params[:request_id]
-    @request = Request.find(request_id)
+    request_id = @message ? @organization.messages.find(params[:id]).request_id : params[:request_id]
```

Comment:
```suggestion
    request_id = @message ? @message.request_id : params[:request_id]
```

if you have already found the message record and saved it to an instance variable and you use that instance variable in the first part of the condition, why not use it to save the request_id, this would save a db call.

Reply:
Ouch .. yes. I really scratched my head over this part and was on it to change it properly .. I think its way better with the last commit now

Reply:
I agree. I think it's an improvement. Thanks :blush: